python_modules/dagster/dagster/_seven/__init__.py

```diff
@@ -142,8 +144,15 @@ def get_current_timestamp():
     return _mockable_get_current_timestamp()
 
 
+def create_datetime(year, month, day, *args, **kwargs):
+    tz = kwargs.pop("tz", "UTC")
+    if isinstance(tz, str):
+        tz = timezone_from_string(tz)
+    return datetime(year, month, day, *args, **kwargs, tzinfo=tz)
```

Comment:
Wow surprised it was this simple with fold. I thought there was something particular about pendulum dst_rule that handled constructing non-existent datetimes which fold would not handle. Do you mind sanity checking that? I don't know if it's worth supporting if all the tests pass though.